src/linear_algebra/matrix.lean

```diff
@@ -257,13 +257,13 @@ linear_equiv_matrix_comp hb hb hb f g
 
 section trace
 
-variables {R : Type v} {M : Type w} [ring R] [add_comm_group M] [module R M]
+variables {R : Type v} {M : Type w} [semiring R] [add_comm_monoid M] [semimodule R M]
 
 /--
 The diagonal of a square matrix.
 -/
 def diag (n : Type u) (R : Type v) (M : Type w)
-  [ring R] [add_comm_group M] [module R M] [fintype n] : (matrix n n M) →ₗ[R] n → M :=
+  [semiring R] [add_comm_monoid M] [semimodule R M] [fintype n] : (matrix n n M) →ₗ[R] n → M :=
```

Comment:
Why not just omit all the arguments here which are already provided as `variables`? You can use `variables (R M)` to make the explicit.